app/Platform/Actions/MigrateAchievementIdsToDifferentGameId.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Platform\Actions;
+
+use App\Community\Enums\ArticleType;
+use App\Models\Achievement;
+use App\Models\Game;
+use App\Models\PlayerAchievement;
+use App\Models\PlayerGame;
+use App\Models\User;
+use App\Platform\Jobs\UpdateGameMetricsJob;
+use App\Platform\Jobs\UpdatePlayerGameMetricsJob;
+use Illuminate\Support\Str;
+
+class MigrateAchievementIdsToDifferentGameId
+{
+    public function execute(array $achievementIds, int $gameId, User $user): void
+    {
+        // Determine which game(s) the achievements are coming from.
+        $oldGameIds = Achievement::whereIn('ID', $achievementIds)->select(['GameID'])->distinct()->pluck('GameID');
+
+        // Associate the achievements to the new game.
+        Achievement::whereIn('ID', $achievementIds)->update(['GameID' => $gameId]);
+
+        // Add an audit comment to the new game.
+        addArticleComment(
+            'Server',
+            ArticleType::GameModification,
+            $gameId,
+            "$user migrated " . Str::plural('achievement', count($achievementIds)) . ' ' .
```

Comment:
Should be `{$user->display_name}`:
```
{"ID":...} migrated achievements 458393,458394,458392 from games 29978,4068.
```

Reply:
But wouldn't it be better to put the GameID modifications in the audit logs for the individual achievements?

Reply:
Applied both in latest. GameID changes are now correctly recorded in the audit log.